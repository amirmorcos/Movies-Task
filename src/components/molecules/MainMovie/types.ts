import { Movie } from "models/Movie";

export type MainMovieProps = {
  movie: Movie;
};
