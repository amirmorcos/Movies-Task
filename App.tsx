/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 */

import { NavigationContainer } from "@react-navigation/native";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import BottomNavigation from "navigation/BottomNavigation";
import React from "react";

const queryClient = new QueryClient();

function App(): React.JSX.Element {
  return (
    <QueryClientProvider client={queryClient}>
      <NavigationContainer>
        <BottomNavigation />
      </NavigationContainer>
    </QueryClientProvider>
  );
}

export default App;
