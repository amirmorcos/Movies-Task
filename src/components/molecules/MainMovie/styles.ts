import { StyleSheet } from "react-native";
import { moderateScale, vs } from "react-native-size-matters";

export default StyleSheet.create({
  imageBg: {
    flex: 1,
    width: undefined,
    height: undefined,
    opacity: 0.8,
    justifyContent: "flex-end",
    alignItems: "center",
  },
  info: {
    color: "#FFC100",
    fontSize: moderateScale(13),
    fontWeight: "bold",
  },
  infoContainer: {
    flexDirection: "row",
    marginBottom: vs(5),
  },
});
