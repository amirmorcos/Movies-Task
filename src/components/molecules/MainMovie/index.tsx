import { useMovie } from "hooks/useMovie";
import React from "react";
import { Text, View } from "react-native";
import styles from "./styles";
import { MainMovieProps } from "./types";
import FastImage from "react-native-fast-image";

const MainMovie = ({ movie }: MainMovieProps) => {
  const { getMoviePoster } = useMovie();
  return (
    <FastImage
      source={{
        uri: getMoviePoster(movie.poster_path),
      }}
      style={styles.imageBg}
      resizeMode="stretch"
    >
      <View style={styles.infoContainer}>
        <Text style={styles.info}>
          {new Date(movie.release_date).getFullYear()}
        </Text>
        <Text style={styles.info}>â€¢</Text>
        <Text style={styles.info}>Action, Romance</Text>
      </View>
    </FastImage>
  );
};

export default MainMovie;
