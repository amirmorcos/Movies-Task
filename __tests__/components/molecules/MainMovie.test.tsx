import { QueryClientProvider } from "@tanstack/react-query";
import { queryClient } from "../../../App";
import React from "react";
import renderer from "react-test-renderer";
import MainMovie from "molecules/MainMovie";

jest.mock("react-native-vector-icons/FontAwesome6", () => "Icon");
jest.mock("react-native-vector-icons/FontAwesome", () => "Icon");
jest.mock("react-native-size-matters", () => ({
  s: jest.fn(),
  vs: jest.fn(),
  moderateScale: jest.fn(),
}));
jest.mock("@react-navigation/bottom-tabs", () => ({
  createBottomTabNavigator: jest.fn(),
}));
jest.mock("react-native-fast-image", () => "FastImage");

const mainMovieObj = {
  adult: false,
  backdrop_path: "/dcnSWFCtk4b2aIzkhq6IDbzoIf1.jpg",
  genre_ids: [28, 35],
  id: 942047,
  original_language: "en",
  original_title: "Outsource",
  overview:
    "A police chief hires an old friend, who is an international spy, to help him search for a wanted suspect in the Philippines. When the chief dies, all evidence points towards the spy, and he must go to extremes to defend himself.",
  popularity: 534.58,
  poster_path: "/zIAF0UXtCJTJOYNYWiBfyifaaOi.jpg",
  release_date: "2022-01-18",
  title: "Outsource",
  video: false,
  vote_average: 2.273,
  vote_count: 11,
};

test("renders correctly", () => {
  const tree = renderer
    .create(
      <QueryClientProvider client={queryClient}>
        <MainMovie movie={mainMovieObj} />
      </QueryClientProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
