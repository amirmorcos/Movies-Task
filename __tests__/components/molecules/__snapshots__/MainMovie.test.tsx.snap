// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "overflow": "hidden",
      },
      {
        "alignItems": "center",
        "flex": 1,
        "height": undefined,
        "justifyContent": "flex-end",
        "opacity": 0.8,
        "width": undefined,
      },
    ]
  }
>
  <FastImageView
    defaultSource={null}
    resizeMode="stretch"
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    style={
      {
        "flexDirection": "row",
        "marginBottom": undefined,
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFC100",
          "fontSize": undefined,
          "fontWeight": "bold",
        }
      }
    >
      2022
    </Text>
    <Text
      style={
        {
          "color": "#FFC100",
          "fontSize": undefined,
          "fontWeight": "bold",
        }
      }
    >
      â€¢
    </Text>
    <Text
      style={
        {
          "color": "#FFC100",
          "fontSize": undefined,
          "fontWeight": "bold",
        }
      }
    />
  </View>
</View>
`;
